lib/remark/heading-ids.ts

```diff
@@ -0,0 +1,61 @@
+import { visit } from 'unist-util-visit';
+import getSectionId from '@/lib/section-id';
+
+type HeaderNodeData = {
+  value?: string;
+  children?: HeaderNodeData[];
+  type: string;
+  data?: {
+    id?: string;
+    hProperties?: {
+      id?: string;
+    };
+  };
+};
+
+type HeadingData = {
+  id?: string;
+  hProperties?: {
+    id?: string;
+  };
+};
+
+/**
+ * Transforms all h tags to add an id based on the text value of the heading
+ *
+ * @returns {HeaderNodeData} The updated node data with the appropriate id attributes
+ */
+const setHeadingIds = () => {
```

Comment:
This will add an id to _every_ heading in the markdown. I don't seem to get the context of what the heading tag is :/

But the titles are all only generated from H2s in `page-info` so this results in just a little bit of extra tagging that does nothing.

Reply:
Is that not what `depth` is?

```
children: [{…}]
data: {hProperties: {…}, id: 'heading-1--introduction'}
depth: 2
position: {start: {…}, end: {…}}
```

Reply:
e.g. if we wanted to:

```
if (!node.children || node.depth !== 2) {
  return;
}
```